libr/anal/p/anal_x86_cs.c

```diff
@@ -986,7 +986,7 @@ static void anop_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len,
 			} else {
 				esilprintf (op,
 					"%s,%s,==,$z,zf,:=,%d,$b,cf,:=,$p,pf,:=,%d,$s,sf,:=,%s,0x%"PFMT64x",-,!,%d,$o,^,of,:=,3,$b,af,:=",
-					src, dst, bitsize, bitsize - 1, src, (1ULL << (bitsize - 1)), bitsize - 1);
+					src, dst, bitsize, bitsize - 1, src, bitsize ? 1ULL << (bitsize - 1) : 0, bitsize - 1);
```

Comment:
`bitsize-1` is used also as the last argument and it would be 0xfffffff if bitsize is 0. I'm not sure what this code does, but this seems wrong.

Reply:
the original esil apparently worked as expected despite its wrongness, but i guess it's better to have no esil at all if `bitsize` is invalid (44eef896a1f0c3da2e626185332cab983a97c074) since it becomes clearer that something needs to be done

Reply:
`bitsize` invalid here means `bitsize` > 64